### ISSUE TITLE
[typescript-angular2] Encodings differ by OS

### ISSUE THREAD

**Issue by Grauenwolf** [swagger-api/swagger-codegen#6971]
##### Description

When generating typescript/angular 2 files on OS X we get a different file encoding than when generating the same files on Windows 10. This is causing a lot of false positives when performing a diff using Visual Studio and git.

##### Swagger-codegen version

2.2.3

##### Command line used for generation

java swagger-codegen generate -i swagger.json -l typescript-angular2 -o ./src/app/proxy/

##### Suggest a fix/enhancement

1. Default encoding that is the same for all operating systems.
2. Generate option to override the default encoding when necessary.

**Comment by macjohnny** [comment]
@Grauenwolf are you referring to the different line endings? in that case you can configure git accordingly: https://help.github.com/articles/dealing-with-line-endings/

**Comment by Grauenwolf** [comment]
No, the presence or lack of a byte order mark. Those annoying, invisible leading characters that denote if a file is ASCII or UTF8.

**Comment by macjohnny** [comment]
@wing328 I guess this would be an issue independent of the language, i.e. the BOM will currently be os-dependent for any target language, isn't it?

**Comment by Grauenwolf** [comment]
Looks like Java picks a default from the OS if the developer doesn't specify. https://stackoverflow.com/questions/9312816/java-platforms-default-charset-on-different-platforms
